### PR TITLE
Optimizer - add missing supported version for BiasSoftmaxFusion

### PR DESCRIPTION
**Pull request by ytaous**
**Description**: Describe your changes.
When testing Babel's TNLRv5-XXL model, checked onnx and found there's missing fusion for Add+Softmax.
Checked the code, and found the version check for softmax node is missing version 13.

**Motivation and Context**

image

### CODE REVIEW

File: onnxruntime/core/optimizer/bias_softmax_fusion.cc [hunk 1]
```diff
@@ -61,7 +61,7 @@ bool TryBiasSoftmaxSubgraphMatch(Graph& graph, Node& start, Node*& add, Node*& s
 
   // check add is only consumed by softmax with matching exec provider
   Node& softmax_node = *graph.GetNode(add_node.OutputNodesBegin()->Index());
-  if (!graph_utils::IsSupportedOptypeVersionAndDomain(softmax_node, "Softmax", {1, 11}) ||
+  if (!graph_utils::IsSupportedOptypeVersionAndDomain(softmax_node, "Softmax", {1, 11, 13}) ||
```
Comment (askhade) [inline onnxruntime/core/optimizer/bias_softmax_fusion.cc:64]: I dont think this is so straightforward. There were multiple changes for this op in version 13.
1. op changed from primitive op to function - should be OK
2. a new bfloat16 type was added - should be OK
3. default value for axis attr changed - more changes are needed to accommodate this change

Please also add relevant tests

Comment (ytaous) [reply onnxruntime/core/optimizer/bias_softmax_fusion.cc:64]: will check again, thx

Comment (askhade) [reply onnxruntime/core/optimizer/bias_softmax_fusion.cc:64]: regarding bfloat16 type addition - initially I said it should be OK... but please check whether the fused op supports this type... if not then you will need a check around this as well .i.e. to reject fusion if original op has bfloat16 type

Comment (askhade) [reply onnxruntime/core/optimizer/bias_softmax_fusion.cc:64]: regarding bfloat16 type addition - initially I said it should be OK... but please check whether the fused op supports this type... if not then you will need a check around this as well .i.e. to reject fusion if original op has bfloat16 type

Comment (ytaous) [reply onnxruntime/core/optimizer/bias_softmax_fusion.cc:64]: You are right, bf16 has not been supported by FusedBiasSoftmax yet, will add type check around original op.  
Good catch.

File: onnxruntime/core/optimizer/bias_softmax_fusion.cc [hunk 2]
```diff
@@ -59,9 +59,23 @@ bool TryBiasSoftmaxSubgraphMatch(Graph& graph, Node& start, Node*& add, Node*& s
     return false;
   }
 
+  // BiasSoftmax supports only float/float16/double - see ./onnxruntime/core/graph/contrib_ops/contrib_defs.cc
+  auto data_type1 = input1->TypeAsProto()->tensor_type().elem_type();
+  if (data_type1 != ONNX_NAMESPACE::TensorProto_DataType_DOUBLE &&
```
Comment (askhade) [inline onnxruntime/core/optimizer/bias_softmax_fusion.cc:64]: nit:  given both the inputs use the same type constraint T so they will always have the same check - you can add a lambda or an inline function for this check and remove the duplicate check on line 70.

**Review by askhade — APPROVED** [review]
LGTM

**Review by hanbitmyths — APPROVED** [review]

**Pull request merged**